lib/ex_plasma/encoding.ex

```diff
@@ -24,11 +25,10 @@ defmodule ExPlasma.Encoding do
 
   # Creates a Merkle proof that transaction under a given transaction index
   # is included in block consisting of hashed transactions
-  @spec merkle_proof(list(String.t()), non_neg_integer()) :: binary()
-  def merkle_proof(hashed_txs, txindex) do
-    build(hashed_txs)
+  @spec merkle_proof(list(binary()), non_neg_integer()) :: binary()
+  def merkle_proof(encoded_transactions, txindex) do
+    build(encoded_transactions)
```

Comment:
it's a good practice not to start a pipiline with a function call, but with a value
```
encoded_transactions
|>build()
|>prove()
|> reverse
|> join ...
```